### ISSUE TITLE
The documentation badly needs an optional arguments section

### ISSUE THREAD

**Issue by JorelAli** [JorelAli/CommandAPI#387]
### Description

Too many people ask about optional arguments and the documentation doesn't help at all! Searching for this in the documentation points to a misleading and unhelpful:
<img width="584" alt="image" src="https://user-images.githubusercontent.com/4613171/209317435-0f3d491f-5a55-457d-9e8c-960d26a49775.png">

### Expected code

```java
new CommandAPICommand("mycommand")
  .executes(...)
  .register();
```

```java
new CommandAPICommand("mycommand")
  .withArguments(new PlayerArgument("optional"))
  .executes(...)
  .register();
```

And CommandTrees:

```java
new CommandTree("mycommand")
    .executes(...)
    .then(new PlayerArgument("optional")
        .executes(...)
    )
    .register();
```

**Comment by willkroboth** [comment]
This [section in the documentation](https://commandapi.jorel.dev/8.7.0/arguments.html#optionaldifferent-arguments) mentions creating optional arguments. It doesn't show up until the bottom of the list though because the title is `Optional/Different Arguments`.
<img width="822" alt="Screen Shot 2022-12-23 at 5 28 14 PM" src="https://user-images.githubusercontent.com/46540330/209368161-71ad19c8-fc3f-4240-91d1-f84925d31f89.png">

The CommandTree page also includes an [example](https://commandapi.jorel.dev/8.7.0/commandtrees.html#example---declaring-a-command-tree-with-a-single-argument) with an optional argument, but it doesn't call it that, so it's not easy to find either.

Adding on to this issue, it would probably be useful to reorganize these examples under a single unified section with a clear name.

**Comment by DerEchtePilz** [comment]
This will probably be added alongside with PR #393 which adds optional arguments.

**Comment by JorelAli** [comment]
Implemented in 9.0.0.
